src/emqx_tls_lib.erl

```diff
@@ -170,8 +172,36 @@ drop_tls13(SslOpts0) ->
         Ciphers -> replace(SslOpts1, ciphers, Ciphers -- ?TLSV13_EXCLUSIVE_CIPHERS)
     end.
 
+inject_root_fun(Options) ->
+    case proplists:get_value(ssl_options, Options) of
+        undefined ->
+            Options;
+        SslOpts ->
+            replace(Options, ssl_options, opt_partial_chain(SslOpts))
+    end.
+
+%% @doc enable TLS partial_chain validation if set.
+-spec opt_partial_chain(SslOpts :: proplists:proplist()) -> NewSslOpts :: proplists:proplist().
+opt_partial_chain(SslOpts) ->
+    case proplists:get_value(partial_chain, SslOpts, undefined) of
+        undefined ->
+            SslOpts;
+        false ->
+            SslOpts;
+        V when V =:= cacert_from_cacertfile orelse V == true ->
+            replace(SslOpts, partial_chain, cacert_from_cacertfile(SslOpts))
+    end.
+
 replace(Opts, Key, Value) -> [{Key, Value} | proplists:delete(Key, Opts)].
 
+%% @doc Helper, make TLS root_fun
+cacert_from_cacertfile(SslOpts) ->
```

Comment:
we need some error handling for this function.
if the cacertfile is not found, or failed to read, or the file is empty, or failed to pem-decode etc.
a big try catch should be good enough, but need a error level lot to help user or supporter to understand what went wrong.

Reply:
added in 
https://github.com/emqx/emqx/pull/10553/commits/31e997c2713a47d2c230b64f42aa07a6b537ba2a

Reply:
we need to throw ST as well